internal/api/client.go

```diff
@@ -61,6 +61,35 @@ var retryableStatusCodes = []int{
 	http.StatusGatewayTimeout,      // 504 is retryable because the server may be overloaded.
 }
 
+// CaePolicyViolationError represents an error when a CAE policy violation is detected.
+type CaePolicyViolationError struct {
+	Message    string
+	StatusCode int
+	Headers    http.Header
+}
+
+func (e *CaePolicyViolationError) Error() string {
+	return fmt.Sprintf("Continuous Access Evaluation policy violation: %s", e.Message)
+}
+
+// IsCaeChallengeResponse detects if a response contains a CAE challenge.
+func IsCaeChallengeResponse(resp *http.Response) bool {
+	if resp == nil {
+		return false
+	}
+
+	// Check response headers for CAE challenge indicators.
+	if resp.StatusCode == http.StatusUnauthorized {
+		wwwAuthenticate := resp.Header.Get("WWW-Authenticate")
+		if wwwAuthenticate != "" {
+			return strings.Contains(wwwAuthenticate, "claims") &&
+				strings.Contains(wwwAuthenticate, "access_denied")
```

Comment:
Maybe "invalid_claim"?

Reply:
Should be "insufficient_claims"

https://learn.microsoft.com/en-us/entra/identity-platform/app-resilience-continuous-access-evaluation?tabs=dotnet#handling-cae-in-your-application